firmware/controllers/actuators/idle_thread.cpp

```diff
@@ -79,8 +79,13 @@ IIdleController::Phase IdleController::determinePhase(int rpm, int targetRpm, Se
 	return Phase::Idling;
 }
 
-float IdleController::getCrankingTaperFraction() const {
-	return (float)engine->rpmCalculator.getRevolutionCounterSinceStart() / engineConfiguration->afterCrankingIACtaperDuration;
+float IdleController::getCrankingTaperFraction(float clt) const {
+    if (engineConfiguration->overrideCrankingTaperDurationSetting) {
+        return (float) engine->rpmCalculator.getRevolutionCounterSinceStart() /
+               (engineConfiguration->afterCrankingIACtaperDuration *
+                interpolate2d(clt, config->cltCrankingTaperCorrBins, config->cltCrankingTaperCorr));
+    }
+    return (float)engine->rpmCalculator.getRevolutionCounterSinceStart() / engineConfiguration->afterCrankingIACtaperDuration;
```

Comment:
I would avoid duplicating this code if you can - it's not obvious to the reader why these two cases are different when they look so similar on first glance.

prefer something like

```c++
float x = config.taperDuration;

if (new_behavior) {
    x *= table_lookup();
}

return getRevCounter() / x;
```

Reply:
that style of writing it makes it very clear which parts are the same, and which parts are different when the setting is enabled